Formula/snappy.rb

```diff
@@ -17,11 +17,28 @@ class Snappy < Formula
   depends_on "cmake" => :build
   depends_on "pkg-config" => :build
 
+  # Fix issue where Mojave clang enters a __GNUC__ block
+  if MacOS.version <= :mojave
+    depends_on "gcc"
+    fails_with :clang do
+      cause "error: invalid output constraint '=@ccz' in asm"
+    end
```

Comment:
Can we use `llvm` Clang instead of `gcc` here?

We don't want `folly` to depend on both `llvm` and `gcc` on Mojave. Instead, it should depend on only one of those, and `folly` can't use `gcc` because it `depends_on "boost"`.

Reply:
Do we need `clang` build version logic, or is swapping out `gcc => llvm` sufficient?

Reply:
If it's a failure tied to the version of Clang used, then we should use `clang_build_version` (as someone might be using an outdated CLT/Xcode to build).

If it's tied to the OS version then we can just `s/gcc/llvm`, but we'd also need to call `ENV.llvm_clang` in the `install` method (since the default fallback is `gcc`).

Reply:
I don't know what is causing failure.
It looks like MacPorts saw failures from 10.8-10.14: https://trac.macports.org/ticket/62954
and just decided to disable all XCode `clang`.

It sounds more likely to be due to a `clang`  version issue and we can try dropping `<=1100` similar to `folly`.

Reply:
Sure, let's do that.